pkg/operator/ceph/csi/ceph_config.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	configName = "csi-ceph-conf-override"
+
+	cephCongKey = "ceph.conf"
+	cephConfVal = `[global]
+auth_cluster_required = cephx
+auth_service_required = cephx
+auth_client_required = cephx
+
+# Workaround for http://tracker.ceph.com/issues/23446
+fuse_set_user_groups = false
+
+# ceph-fuse which uses libfuse2 by default has write buffer size of 2KiB
+# adding 'fuse_big_writes = true' option by default to override this limit
+# see https://github.com/ceph/ceph-csi/issues/1928
+fuse_big_writes = true
+`
+	// keyring should be adds to the same configmap as ceph.conf as we cannot
+	// create keyring file in the same directory as ceph.conf.
+	keyringKey = "keyring"
+	// empty keyring file content
+	keyringVal = ""
+)
+
+// CreateCsiConfigMap creates an ceph.conf for ceph-csi. If a config map already
+// exists, it will return it.
+func CreateCsiCephConfigMap(ctx context.Context, clientset kubernetes.Interface, ownerInfo *k8sutil.OwnerInfo, namespace string) error {
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configName,
+			Namespace: namespace,
+		},
+	}
+	configMap.Data = map[string]string{
+		cephCongKey: cephConfVal,
+		keyringKey:  keyringVal,
```

Comment:
Why add an empty keyring to the configmap? Do you expect users to update with the real keyring? Or why is this needed? Seems like it should be in a secret if anywhere.

Reply:
we don't use this keyring for csi operations, but if this file is not created we see a warning when we execute ceph CLI commands. that's the reason for having an empty file.